chain/bitcoind_events.go

```diff
@@ -0,0 +1,523 @@
+			// bitcoind shuts down, which will produce an unreadable
+			// event type. To prevent from logging it, we'll make
+			// sure it conforms to the ASCII standard.
+			if eventType == "" || !isASCII(eventType) {
+				continue
+			}
+
+			log.Warnf("Received unexpected event type from %v "+
+				"subscription: %v", rawTxZMQCommand, eventType)
+		}
+	}
+}
+
+func (b *BitcoindZMQEvents) Stop() error {
+	var returnErr error
+	if err := b.txConn.Close(); err != nil {
+		returnErr = err
+	}
+
+	if err := b.blockConn.Close(); err != nil {
+		returnErr = err
+	}
+
+	close(b.quit)
+	b.wg.Wait()
+	return returnErr
+}
+
+type PollingConfig struct {
+	BlockPollingTime   time.Duration
+	TxPollingTime      time.Duration
+	MempoolEvictionAge time.Duration
+}
+
+type BitcoindRPCPollingEvents struct {
+	cfg *PollingConfig
+
+	client *rpcclient.Client
+
+	mempool *mempool
+
+	notifyBlock func(block *wire.MsgBlock)
+	notifyTx    func(tx *wire.MsgTx)
+
+	wg   sync.WaitGroup
+	quit chan struct{}
+}
+
+func newBitcoindRPCPollingEvents(cfg *PollingConfig, client *rpcclient.Client,
+	notifyBlock func(block *wire.MsgBlock),
+	notifyTx func(tx *wire.MsgTx)) (*BitcoindRPCPollingEvents, error) {
+
+	return &BitcoindRPCPollingEvents{
+		cfg:         cfg,
+		client:      client,
+		notifyTx:    notifyTx,
+		notifyBlock: notifyBlock,
+		mempool:     newMempool(),
+		quit:        make(chan struct{}),
+	}, nil
+}
+
+func (b *BitcoindRPCPollingEvents) Start() error {
+	info, err := b.client.GetBlockChainInfo()
+	if err != nil {
+		return err
+	}
+
+	b.wg.Add(3)
+	go b.blockEventHandlerRPC(info.Blocks)
+	go b.txEventHandlerRPC()
+	go b.evictOldTransactions()
+
+	return nil
+}
+
+func (b *BitcoindRPCPollingEvents) Stop() error {
+	close(b.quit)
+	b.wg.Wait()
+	return nil
+}
+
+// blockEventHandlerRPC is a goroutine that uses the rpc client to check if we
+// have a new block every so often.
+func (b *BitcoindRPCPollingEvents) blockEventHandlerRPC(startHeight int32) {
+	defer b.wg.Done()
+
+	log.Info("Started polling for new bitcoind blocks via RPC.")
+	ticker := time.NewTicker(b.cfg.BlockPollingTime)
+	defer ticker.Stop()
+
+	height := startHeight
+	log.Info("startheight: ", height)
+
+	for {
+		select {
+		case <-ticker.C:
+			// Every so many seconds, we poll to see if there's a
+			// new block.
+			info, err := b.client.GetBlockChainInfo()
+			if err != nil {
+				log.Errorf("Unable to retrieve best block: "+
+					"%v", err)
+				continue
+			}
+
+			// If the block isn't new, we continue. Else, we need
+			// grab the full block data to send to the clients.
+			if info.Blocks <= height {
+				continue
+			}
+
+			hash, err := chainhash.NewHashFromStr(info.BestBlockHash)
+			if err != nil {
+				log.Errorf(err.Error())
+				return
+			}
+
+			for i := height + 1; i <= info.Blocks; i++ {
+				newBlock, err := b.client.GetBlock(hash)
+				if err != nil {
+					log.Errorf("Unable to retrieve "+
+						"block: %v", err)
+					continue
+				}
+
+				b.notifyBlock(newBlock)
```

Comment:
I'm not seeing this notifyBlock function

Reply:
it is passed in as a config value. So it is actually the `sendBlockToClients` function